tokio-util/src/time/wheel/mod.rs

```diff
@@ -116,9 +116,17 @@ where
         Ok(())
     }
 
-    /// Remove `item` from thee timing wheel.
+    /// Remove `item` from the timing wheel.
     pub(crate) fn remove(&mut self, item: &T::Borrowed, store: &mut T::Store) {
         let when = T::when(item, store);
+
+        assert!(
```

Comment:
By the way, I added this assert because while the assert in `level_for` wasn't valid in general, it turned out to be helpful in surfacing some forms of inconsistent state; that's how issue #2473 (looking up entries in the wheel which had been moved to the expired queue) manifested. Asserting this invariant can catch at least some cases of invalid API use or delay queue corruption.

I added the same condition as a `debug_assert` in the main Tokio crate, where I don't think it could be hit other than due to corrupt internal state.